pandas/tseries/index.py

```diff
@@ -1202,9 +1202,10 @@ def intersection(self, other):
               not other.offset.isAnchored() or
               (not self.is_monotonic or not other.is_monotonic)):
             result = Index.intersection(self, other)
-            if isinstance(result, DatetimeIndex):
-                if result.freq is None:
-                    result.offset = to_offset(result.inferred_freq)
+            result = self._simple_new(result._values, name=result.name,
```

Comment:
no, use ``_shallow_copy`` with ``freq=None``, which will not infer it, and then you can reset the freq after